CONTRIBUTING.md

```diff
@@ -0,0 +1,196 @@
+
+## Table of Contents
+
+- [Project Architecture](#project-architecture)
+  - [Some Important Components](#some-important-components)
+    - [Background Page](#background-page)
+    - [Popup Page](#popup-page)
+    - [Activity Log Page - Tab](#activity-log-page---tab-extension-page)
+    - [Activity Log Page - Devtools](#activity-log-page---devtools-extension-page)
+  - [Core Features](#core-features)
+    - [Collecting Logs](#collecting-logs)
+      - [Live Logging](#live-logging)
+      - [Loading / saving logs](#loading--saving-logs)
+    - [Rendering Logs](#rendering-logs)
+    - [Filtering Log Entries](#filtering-log-entries)
+- [Picking an issue](#picking-an-issue)
+- [Installation](#installation)
+- [Writing and Running Tests](#writing--running-tests)
+  - [Writing a test](#writing-a-test)
+- [Checking for Linter](#checking-for-linter)
+- [Checking for Prettier](#checking-for-prettier)
+- [Submitting a Bug or Issue or Feature Request](#submitting-a-bug-or-issue-or-feature-request)
+- [Creating a Pull Request](#creating-a-pull-request)
+  - [Writing Commit Message](#writing-commit-message)
+
+## Project Architecture
+
+Extension Activity Monitor is composed by a backend part (running in the background page) and a frontend part (composed by a browserAction Popup and an Activity Log extension page which is loaded as a tab or as a devtools panel).
+
+### Some Important Components
+
+#### Background Page
+
+The background page acts as the backend of the extension. The core part of the background page is implemented in [`ext-monitor.js`](https://github.com/mozilla/extension-activity-monitor/blob/master/src/lib/ext-monitor.js). The Popup and Activity Log page communicates with the background page via `runtime.onMessage` event and instruct to start/stop monitoring, save logs to a JSON file or load logs from a JSON file. It keeps track of the monitored extensions via [`extensionMapList`](https://github.com/mozilla/extension-activity-monitor/blob/68d51940f1db397a0972658622bbdd39041436a7/src/lib/ext-monitor.js#L9) and opened Activity Log pages via [`activityLogPorts`](https://github.com/mozilla/extension-activity-monitor/blob/68d51940f1db397a0972658622bbdd39041436a7/src/lib/ext-monitor.js#L12).
+
+The logs remain alive in the background unless it receives the [`clearlogs`](https://github.com/mozilla/extension-activity-monitor/blob/68d51940f1db397a0972658622bbdd39041436a7/src/lib/ext-monitor.js#L106) instruction from Activity Log page(frontend part). The background page also [sends the real-time logs](https://github.com/mozilla/extension-activity-monitor/blob/68d51940f1db397a0972658622bbdd39041436a7/src/lib/ext-monitor.js#L25) to Activity Log page.
+
+#### Popup Page
+
+The popup page is one part of the frontend that communicates with the background page (backend part) to run most of its functionality. It is responsible for "start monitoring" and "stop monitoring" other installed extensions. It also helps to access the Activity Log page.
```

Comment:
"Responsible" is not the best description here. The page only triggers these actions, it's not responsible for executing or holding state etc.

Reply:
Rephased it like the following: 
`It triggers "start monitoring" and "stop monitoring" functionalities and helps to access the Activity Log page.`